src/components/ColumnPicker/NewColumnPickerAvailableNodes.tsx

```diff
@@ -0,0 +1,84 @@
+import { CircularProgress } from "@material-ui/core";
+import { Box, Checkbox, Text } from "@saleor/macaw-ui/next";
+import React from "react";
+import { FormattedMessage } from "react-intl";
+
+import messages from "./messages";
+import { NewColumnPickerSearch } from "./NewColumnPickerSearch";
+import { ColumnCategory } from "./useColumns";
+
+export interface NewColumnPickerAvailableNodesProps {
+  currentCategory: ColumnCategory;
+  columnPickerSettings: string[];
+  query: string;
+  setQuery: React.Dispatch<React.SetStateAction<string>>;
+  changeHandler: (column: string) => void;
+}
+
+export const NewColumnPickerAvailableNodes = ({
+  currentCategory,
+  columnPickerSettings,
+  query,
+  setQuery,
+  changeHandler,
+}) => {
+  const areNodesLoading = currentCategory.availableNodes === undefined;
+  const areNodesEmpty = currentCategory.availableNodes?.length === 0;
+
+  const renderNodes = () => {
```

Comment:
**suggestion**: this is basically a component, why do you want to be a function like this?

Reply:
Also you this this entire component when it's split into smaller pieces, can avoid props drilling - please notice that some you require some of the properties just for one use that even is happening during some of conditions (not always)